tests/auto/tst_persistenttabmodel/tst_persistenttabmodel.cpp

```diff
@@ -428,11 +494,45 @@ void tst_persistenttabmodel::onTitleChanged()
 
     QSignalSpy dataChangedSpy(tabModel, SIGNAL(dataChanged(QModelIndex, QModelIndex, QVector<int>)));
 
-    mockPage.m_tabId = 1;
-    mockPage.setTitle("Hello world");
+    EXPECT_CALL(mockPage, tabId()).WillOnce(Return(1));
+    QUrl url;
+    EXPECT_CALL(mockPage, url()).WillOnce(Return(url));
+    EXPECT_CALL(mockPage, title()).WillOnce(Return(QString("Hello world")));
+    emit mockPage.titleChanged();
+
     QCOMPARE(dataChangedSpy.count(), 1);
 }
 
+void tst_persistenttabmodel::nextActiveTabIndex()
+{
+    DeclarativeWebContainer container;
+    DeclarativeWebPage page;
+    tabModel->setWebContainer(&container);
+
+    EXPECT_CALL(container, webPage()).WillRepeatedly(Return(&page));
+    EXPECT_CALL(page, parentId()).WillOnce(Return(1));
+    EXPECT_CALL(page, tabId()).WillOnce(Return(2));
+    tabModel->nextActiveTabIndex(0);
+}
+
+void tst_persistenttabmodel::misc()
+{
+    // Call simple methods to increase coverage
```

Comment:
Agree, we need to add also QML tests.

For now I've split this case to 3 different cases dedicated to their respective methods. The case that simply calls a method is still useful since it makes sure the public method doesn't explode.